src/common/ipc.ts

```diff
@@ -1,79 +1,72 @@
-// Inter-protocol communications (main <-> renderer)
+// Inter-process communications (main <-> renderer)
 // https://www.electronjs.org/docs/api/ipc-main
 // https://www.electronjs.org/docs/api/ipc-renderer
 
-import { ipcMain, ipcRenderer, WebContents, webContents } from "electron"
+import { ipcMain, ipcRenderer, webContents, remote } from "electron"
 import logger from "../main/logger";
 
-export type IpcChannel = string;
+export function handleRequest(channel: string, listener: (...args: any[]) => any) {
+  ipcMain.handle(channel, listener)
+}
 
-export interface IpcChannelOptions {
-  channel: IpcChannel; // main <-> renderer communication channel name
-  handle?: (...args: any[]) => Promise<any> | any; // message handler
-  autoBind?: boolean; // auto-bind message handler in main-process, default: true
-  timeout?: number; // timeout for waiting response from the sender
-  once?: boolean; // one-time event
+export async function requestMain(channel: string, ...args: any[]) {
+  return ipcRenderer.invoke(channel, ...args)
 }
 
-export function createIpcChannel({ autoBind = true, once, timeout = 0, handle, channel }: IpcChannelOptions) {
-  const ipcChannel = {
-    channel: channel,
-    handleInMain: () => {
-      logger.info(`[IPC]: setup channel "${channel}"`);
-      const ipcHandler = once ? ipcMain.handleOnce : ipcMain.handle;
-      ipcHandler(channel, async (event, ...args) => {
-        let timerId: any;
-        try {
-          if (timeout > 0) {
-            timerId = setTimeout(() => {
-              throw new Error(`[IPC]: response timeout in ${timeout}ms`)
-            }, timeout);
-          }
-          return await handle(...args); // todo: maybe exec in separate thread/worker
-        } catch (error) {
-          throw error
-        } finally {
-          clearTimeout(timerId);
-        }
+async function getSubFrames(): Promise<number[]> {
+  const subFrames: number[] = [];
+  const { clusterStore } = await import("./cluster-store");
+  clusterStore.clustersList.forEach(cluster => {
+    if (cluster.frameId) {
+      subFrames.push(cluster.frameId)
+    }
+  });
+  return subFrames;
+}
```

Comment:
Instead of managing `frameId` in cluster/cluster-store, maybe better to handle a `Map<clusterId, frameId` here and re-use in broadcast?

```ts
export const clusterFrameMap = observable.map();

if(ipcMain){
  handleRequest(clusterSetFrameIdHandler, (clusterId: ClusterId, frameId: number) => {
    clusterFrameId.set(clusterId, frameId)
  })
}
```

Reply:
Btw now `cluster-ipc.ts` looks like `ipc-cluster-server` :D

Reply:
> Instead of managing frameId in cluster/cluster-store, maybe better to handle a Map<clusterId, frameId here and re-use in broadcast? 

That's a good idea, refactored and now we also got rid of `Cluster.frameId`.